assets/app/scripts/directives/oscFileInput.js

```diff
@@ -31,6 +49,79 @@ angular.module('openshiftConsole')
           };
           reader.readAsText(file);
         });
+
+        // Specify event handler for the whole page drag & drop actions
+        $(document).on('dragenter', function(e) {
+          preventAndStop(e);
+          if (isDisabled) {
+            return false;
+          }
+          $(dropArea).addClass('show-drag-and-drop-area');
+          showDropArea = true;
+        });
+        $(document).on('dragover', function(e) {
+          preventAndStop(e);
+          if (isDisabled) {
+            return false;
+          }
+          showDropArea = true;
+        });
+        $(document).on('dragleave', function(e) {
+          preventAndStop(e);
+          if (isDisabled) {
+            return false;
+          }
+          showDropArea = false;
+          clearTimeout( timeout );
+          timeout = setTimeout( function(){
+              if( !showDropArea ){ $(dropArea).removeClass('show-drag-and-drop-area'); }
+          }, 200 );
+        });
+        $(document).on('drop', function(e) {
+          preventAndStop(e);
+          return false;
+        });
+
+        // Specify event handler for the dropArea drag & drop actions
+        $(dropArea).on('dragenter', function() {
+          $(dropArea).addClass('highlight-drag-and-drop-area');
+          highlightDropArea = true;
+        });
+        $(dropArea).on('dragover', function() {
+          highlightDropArea = true;
+        });
+        $(dropArea).on('dragleave', function() {
+          highlightDropArea = false;
+          clearTimeout( timeout );
+          highlightTimeout = setTimeout( function(){
+              if( !highlightDropArea ){ $(dropArea).removeClass('highlight-drag-and-drop-area'); }
```

Comment:
Is there a reason we can't just always add the class on `dragenter` and remove the class on `dragleave`?